02_bubble/saurabhvyas9/main_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	// Given
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote("[1 2 3 5][1 2 3 5]")
+	actual := strconv.Quote(buf.String())
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %s\nactual: %s", expected, actual)
+	}
+}
+
+var testCases = map[string]struct {
+	input []int
+	want  []int
+}{
+	"TC1": {input: []int{3, 1, 2, 5}, want: []int{1, 2, 3, 5}},
+	"TC2": {input: []int{1, 2, 4, 3, 5}, want: []int{1, 2, 3, 4, 5}},
+	"TC3": {input: []int{5, 4, 3, 2, 1}, want: []int{1, 2, 3, 4, 5}},
+	"TC4": {input: []int{4, 4, 1, 100, 1}, want: []int{1, 1, 4, 4, 100}},
+	"TC5": {input: []int{-5, -1, -2, 4, 3, 19, 0, 1, 4, 7, 8}, want: []int{-5, -2, -1, 0, 1, 3, 4, 4, 7, 8, 19}},
+}
+
+func TestBubbleSort(t *testing.T) {
+	for _, test := range testCases {
+		input := test.input
+		sorted := bubble(input)
+		if !Equal(sorted, test.want) {
+			t.Errorf("Sorting failed. Input: %v, Sorted: %v.", input, sorted)
+		}
+	}
+}
+
+func TestInsertionSort(t *testing.T) {
+	for _, test := range testCases {
+		input := test.input
+		sorted := insertion(input)
+		if !Equal(sorted, test.want) {
+			t.Errorf("Sorting failed. Input: %v, Sorted: %v.", input, sorted)
+		}
+	}
+}
+
+// Equal tells whether a and b contain the same elements.
+// A nil argument is equivalent to an empty slice.
+func Equal(a, b []int) bool {
```

Comment:
Nicely done - however, `testify/assert` has got and `assert.Equal` that does it for you out of the box with great diff-printing if the test fails.
No need to update here, for the next lab.

Reply:
Done.